app/helpers/lightweight_activity_helper.rb

```diff
@@ -58,4 +58,12 @@ def complete_badge_for(activity)
       return ribbon(t("COMPLETED"),"my-ribbon")
     end
   end
+
+  def activity_player_path(activity)
+    return  "#{ENV['ACTIVITY_PLAYER_URL']}/?activity=#{@activity.id}&baseUrl=#{CGI.escape(root_url)}"
```

Comment:
Does the activity player need to make additional requests back to lara?
My preference would be to keep this launching argument simple. So it just passes the activityUrl directly instead of the baseUrl and the activity id. This way the activity player doesn't have any lara specific knowledge about how to construct the activity url.

You should be able to get the activity url by calling `lightweight_activity_url(@activity)` you might need the id on there: `lightweight_activity_url(@activity.id)`

Reply:
Hmm. Part of my thinking was that we'd eventually want to support urls that are as clean as the ones in existing LARA, such as `/activity/123/`, and so wanted to separate the activity # from the base url in case we set a global base url in the Activity Player at some point. But since this isn't on the radar (and can always be changed in the future) this maybe isn't necessary.

In terms of LARA-specific knowledge, what it uses is the API route, which I figured was supposed to be something freely useable by anything anticipating working with a LARA-like service.

But I can go ahead and change this to use the url directly. `lightweight_activity_url` doesn't seem to be supported, with either argument. In any case what we need is the `api/` url. Do you know if there's a magic rails route that would go to that?

Reply:
In general you can run `rake routes` to find these magic names. In docker from your host that would be the craxy `docker exec app bundle exec rake routes`.

I'd guess based on the route.rb file that what you need is `activity_url(@activity)`

I just did a little digging and we might be able to support urls like activity-player.concord.org/activities/123 using this approach: https://stackoverflow.com/questions/16267339/s3-static-website-hosting-route-all-paths-to-index-html

But yeah I think it should be a different feature since we'll also want previewing from developer machines and staging to work. And with the url parameter we can also create static test exports which the activity player can load in.

Reply:
:+1:

Updated in fb18e72, Activity Player updated in https://github.com/concord-consortium/activity-player/pull/21.